app/main.py

```diff
@@ -1,13 +1,42 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": item["line_number"],
+                "column": item["column_number"],
+                "message": item["text"],
+                "name": item["code"],
+                "source": "flake8"
+            } for item in errors
+        ],
+        "path": file_path,
+        "status": "failed"
+    }
```

Comment:
There is a duplication in your code. You are repeating the same block of code in the `format_single_linter_file` function and inside the `format_linter_report` function. To follow the DRY (Don't Repeat Yourself) principle, you should reuse the `format_linter_error` function inside the `format_single_linter_file` function. This way, you avoid repeating the same logic for formatting linter errors.